src/test.js

```diff
@@ -356,8 +356,8 @@ Test.prototype = {
 	},
 
 	valid: function() {
-		var include,
-			filter = config.filter && config.filter.toLowerCase(),
+		var filter = config.filter,
+			regexFilter = /^!?\/([\w\W]*)\/(i?$)/.exec( filter ),
```

Comment:
this regexp also capture strings starting with `!`. IMHO, we may remove it and go with `/^\/([\w\W]*)\/(i?$)/` only.

The `!` prefix flag is good for is strings but not necessary at all for regexp.

Also the current regexFilter function is not handling this prefix.

Removing the `!` flag for regexps is not a breaking feature as it is still operational for regular string searches.